### DISCUSSION TITLE
🚀 Feature: Twitter auth provider

### DISCUSSION THREAD

**Discussion by byawitz** [appwrite/appwrite#11516] [Ideas]
### 🔖 Feature description

Adding Login with Twitter option in auth. 

I can add the provider.

### 🎤 Pitch

Twitter is popular and has [full support](https://developer.twitter.com/en/docs/authentication/guides/log-in-with-twitter) for OAuth protocols.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by joeyouss** [comment]
Thank you for opening this <3

Adding a related comment here: https://github.com/appwrite/appwrite/issues/410#issuecomment-1030484754

**Comment by byawitz** [comment]
The comment indeed requires extra attention on my side. I will look into it in the beginning of next week. 

Thanks

**Comment by joeyouss** [comment]
Thank you! Let us know if you have any thoughts/feedback

**Comment by joeyouss** [comment]
Seems like Twitter did add more support, this issue can be worked on now probably.

**Comment by yatendra2001** [comment]
Hey @joeyouss, I would like to start working on this issue. Can you assign this to me??

Also, is anybody working on this issue already??

The last thing we want is multiple people working on the same issue :(

**Comment by egemenu** [comment]
Why there is no action on this? :/

**Comment by stnguyen90** [comment]
Unassigned due to inactivity

**Comment by kotatsu-dog** [comment]
@stnguyen90 
Hey!
It looks like Twitter (X) now supports OAuth2, is that right?
If it is right,
I want to try integrating Twitter as an OAuth provider for Appwrite.

https://docs.x.com/resources/fundamentals/authentication/oauth-2-0/authorization-code

**Comment by rwxdevjavu** [comment]
@stnguyen90  I would like to start working on this issue. Can you assign this to me??

**Comment by krish-wp** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?
